### PR TITLE
add search command

### PR DESCRIPTION
**Pull request by aireater**
Create a search command for searching text for every column in every table. It can take a while in a large database.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ command         description
 --------------  ------------------------------------------
 .help           Shows this message
 .tables         Lists all the tables
+.search         Search all columns in all tables for a value (takes a while)
```
Comment (hasankhan) [inline README.md]: please remove (takes a while)

File: lib/commands/search.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    var Utils = require('./utils'),
+        Queries = require('./queries'),
+        Q = require('q');
+
+    class SearchCommand {
+        constructor(db) {
+            this.db = db;
+            this.prefix = '.search';
+            this.usage = '.search';
+            this.description = 'Search all columns in all tables for a value (takes a while)';
```
Comment (hasankhan) [inline lib/commands/search.js]: please remove takes a while

File: lib/commands/search.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        }
+
+        run(messages, writer, args) {
+            if (!Utils.validateArgs(args, 1)) {
+                return Q.reject('No search value supplied');
+            }
+
```
Comment (hasankhan) [inline lib/commands/search.js]: maybe we can write a message to screen before the command runs, something like 'Searching...'
please make sure that the message is only written to console and does not become part of file output

File: lib/commands/search.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+(function () {
+    "use strict";
+
+    var Utils = require('./utils'),
+        Queries = require('./queries'),
+        Q = require('q');
+
+    class SearchCommand {
```
Comment (hasankhan) [inline lib/commands/search.js]: please add a test for search command

File: lib/commands/index.js [hunk 5]
```diff
@@ -19,6 +19,7 @@
         HelpCommand: require('./help'),
```
Comment (hasankhan) [inline lib/commands/index.js]: please update the test for help command and assert that search is also present

**Comment by aireater** [comment]
@hasankhan will do, is there a command i can look at for the "Searching..." console output?

**Comment by hasankhan** [comment]
There isn't an example unfortunately, your command would be the first one to do so. You can use the messages object passed to the method.

**Comment by aireater** [comment]
@hasankhan is the test for the search good enough?

**Pull request merged**